docs/contracts/v4/guides/11-position-manager.mdx

```diff
@@ -0,0 +1,838 @@
+bytes[] memory params = new bytes[](3);
+```
+
+Now, let’s encode parameters for the old position:
+
+```solidity
+// Parameters for BURN_POSITION
+params[0] = abi.encode(
+    oldPositionId,
+    0,  // No minimum for token0 (consider adding slippage protection)
+    0,  // No minimum for token1
+    ""  // No hook data
+);
+```
+
+Then for minting the new position:
+
+```solidity
+// Parameters for MINT_POSITION
+params[1] = abi.encode(
+    newPositionParams.poolKey,
+    newPositionParams.tickLower,
+    newPositionParams.tickUpper,
+    newPositionParams.liquidity,
+    newPositionParams.amount0Max,
+    newPositionParams.amount1Max,
+    address(this),  // New position owner
+    ""             // No hook data
+);
+```
+
+Next, we handle token settlements. First for the new position:
+
+```solidity
+// Parameters for SETTLE_PAIR (providing tokens for new position)
+params[2] = abi.encode(
+  newPositionParams.poolKey.currency0,
+  newPositionParams.poolKey.currency1
+);
+```
+
+With everything prepared, we can execute our batch operation:
+
+```solidity
+// Execute all operations atomically
+    positionManager.modifyLiquidities(
+        abi.encode(actions, params),
+        block.timestamp + 60
+    );
+}
+```
+
+# Delta-Resolving Operations
+
+While we’ve seen basic delta resolution using SETTLE_PAIR and TAKE_PAIR in previous sections, v4’s Position Manager provides additional operations for handling more complex scenarios. Let’s understand when and how to use each one.
+
+## **CLOSE_CURRENCY: Handling Unknown Deltas**
+
+When you can’t predict whether you’ll need to pay or receive tokens, CLOSE_CURRENCY automatically handles either case.
+
+```solidity
+// Example scenario: Converting fees to liquidity
+bytes memory actions = abi.encodePacked(
+    Actions.INCREASE_LIQUIDITY,
+    Actions.CLOSE_CURRENCY  // Will automatically settle or take based on final delta
+);
+
+bytes[] memory params = new bytes[](2);
+
+// Parameters for INCREASE_LIQUIDITY
+params[0] = abi.encode(
+    tokenId,
+    liquidityIncrease,
+    amount0Max,
+    amount1Max,
+    ""
+);
+
+// CLOSE_CURRENCY only needs the currency
+params[1] = abi.encode(currency0);
+```
+
+This is particularly useful when:
+
+- Converting fees to liquidity (fees might fully cover the increase)
+- Complex operations where final deltas are uncertain
+- Reducing code complexity by letting the protocol handle the direction
+
+## **CLEAR_OR_TAKE: Optimizing for Dust**
+
+Sometimes receiving small token amounts costs more in gas than they’re worth. CLEAR_OR_TAKE lets you specify a threshold:
+
+```solidity
+// Parameters for CLEAR_OR_TAKE
+params[0] = abi.encode(
+    currency,     // The token to handle
+    threshold     // Minimum amount worth taking
+);
+```
+
+If the amount to receive is:
+
+- Above threshold: Tokens are taken (like TAKE_PAIR)
+- Below threshold: Amount is forfeited, saving gas
+
+This is valuable for:
+
+- Fee collection with minimum amounts
```

Comment:
lets remove this, doesnt seem right to me that you would just "collect fees" to forfeit immediately?

Reply:
removed